database/postgres/pg.js

```
@@ -0,0 +1,51 @@
+const { Client } = require('pg')
+require('dotenv').config()
+
+const pgModule = {}
+let client
+
+pgModule.startPGDB = async ()=>{
+  try{
+    client = new Client({
+      host: process.env.HOST,
+      port: process.env.PORT,
+      user: process.env.USERNAME,
+      password: process.env.PASSWORD,
+      database: process.env.DATABASE
+    })
+    await client.connect()
+  }catch(err){
+    console.log('connection failed', err)
+  }
+}
+
+pgModule.closePGDB = async ()=>{
+  try{
+    await client.end()
+  }catch(err){
+    console.log('connection failed to closed', err)
+  }
+}
+
+pgModule.createPgAccount = async (username, password)=>{
+  try{
+    if(!username || !password) return
+    await client.query(`CREATE DATABASE IF NOT EXISTS ${username}`)
+    await client.query(`CREATE USER IF NOT EXISTS ${username} WITH ENCRYPTED password '${password}'`)
+    await client.query(`GRANT ALL PRIVILEGES ON DATABASE ${username} TO ${username}`)
+  }catch(err){
+    console.log('failed to createPgAccount', err)
```

please throw an error here:

```
throw new Error(`failed to createPGAccount for user: ${username}`)
```

make sure your tests address this issue.